407 Improve Ticket Labels

Mecktro opened the pull request:
This PR updates how severity labels are assigned to DevGuard-created tickets. Previously, each ticket received a `devguard` label along with a `severity:risk` label, where severity was determined solely by risk, not the CVSS score.  

### Changes Made:  
- Updated the labeling logic in `github_integration.go` and `gitlab_integration.go` (located in `devguard/internal/core/integrations`).  
- Labels now include:  
  - **Risk-based severity:** `"risk:" + strings.ToLower(risk.RiskToSeverity(vuln.GetRawRiskAssessment()))`  
  - **CVSS-based severity:** `"cvss-severity:" + strings.ToLower(risk.RiskToSeverity(vuln.GetRawRiskAssessment()))`  

### Expected Output:  
- Example labels:  
  - `"risk:high"`  
  - `"cvss-severity:high"`  

### Related Issue:  
[[Issue #407](https://github.com/l3montree-dev/devguard/issues/407)](https://github.com/l3montree-dev/devguard/issues/407)

timbastin commented:
Hi @Mecktro thanks a lot for your contribution! The vulnerability object does actually store two different values: The risk (gets calculated based on the cvss, epss vulnerability depth etc.) and the cvss basescore. It would be great to have both information as labels. Currently you are adding the same value, the risk, to both labels.

I think you need something like this:

```go
labels := []string{
  "devguard",
  "risk:" + strings.ToLower(risk.RiskToSeverity(vuln.GetRawRiskAssessment())), // using the calculated risk
  "state:accepted",
}

if v, ok := vuln.(*models.DependencyVuln); ok {
  labels = append(labels, "cvss-severity:"+risk.RiskToSeverity(float64(v.CVE.CVSS))) // using the cvss value
}
```

Are you willing to-do that change? Maybe you can even create a function to get the matching labels.

```go
getLabels(vuln models.Vuln, state string) []string
```

Mecktro commented:
yes thanks for correcting and I will change it as you said

On Sun, Mar 30, 2025 at 2:47 PM Tim Bastin ***@***.***> wrote:

> Hi @Mecktro <https://github.com/Mecktro> thanks a lot for your
> contribution! The vulnerability object does actually store two different
> values: The risk (gets calculated based on the cvss, epss vulnerability
> depth etc.) and the cvss basescore. It would be great to have both
> information as labels. Currently you are adding the same value, the risk,
> to both labels.
>
> I think you need something like this:
>
> labels := []string{
>   "devguard",
>   "risk:" + strings.ToLower(risk.RiskToSeverity(vuln.GetRawRiskAssessment())), // using the calculated risk
>   "state:accepted",
> }
> if v, ok := vuln.(*models.DependencyVuln); ok {
>   labels = append(labels, "cvss-severity:"+risk.RiskToSeverity(float64(v.CVE.CVSS))) // using the cvss value
> }
>
> Are you willing to-do that change? Maybe you can even create a function to
> get the matching labels.
>
> getLabels(vuln models.Vuln, state string) []string
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/l3montree-dev/devguard/pull/408#issuecomment-2764466862>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARSKQBSICBMW5DCUJSSSNSD2W6ZENAVCNFSM6AAAAAB2CFN35CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRUGQ3DMOBWGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: timbastin]*timbastin* left a comment (l3montree-dev/devguard#408)
> <https://github.com/l3montree-dev/devguard/pull/408#issuecomment-2764466862>
>
> Hi @Mecktro <https://github.com/Mecktro> thanks a lot for your
> contribution! The vulnerability object does actually store two different
> values: The risk (gets calculated based on the cvss, epss vulnerability
> depth etc.) and the cvss basescore. It would be great to have both
> information as labels. Currently you are adding the same value, the risk,
> to both labels.
>
> I think you need something like this:
>
> labels := []string{
>   "devguard",
>   "risk:" + strings.ToLower(risk.RiskToSeverity(vuln.GetRawRiskAssessment())), // using the calculated risk
>   "state:accepted",
> }
> if v, ok := vuln.(*models.DependencyVuln); ok {
>   labels = append(labels, "cvss-severity:"+risk.RiskToSeverity(float64(v.CVE.CVSS))) // using the cvss value
> }
>
> Are you willing to-do that change? Maybe you can even create a function to
> get the matching labels.
>
> getLabels(vuln models.Vuln, state string) []string
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/l3montree-dev/devguard/pull/408#issuecomment-2764466862>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARSKQBSICBMW5DCUJSSSNSD2W6ZENAVCNFSM6AAAAAB2CFN35CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRUGQ3DMOBWGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

timbastin commented:
I made some minor changes already to the ticket generation. Therefore, there is a merge conflict right now. I think it should be pretty simple to solve, since I just moved the ticket generation into their functions. If you need help, just reach out again.

Mecktro commented:
> I made some minor changes already to the ticket generation. Therefore, there is a merge conflict right now. I think it should be pretty simple to solve, since I just moved the ticket generation into their functions. If you need help, just reach out again.

Thanks for the heads-up! I’ve rebased the changes and created a common `getLabels` function to dynamically generate labels. Let me know if any further updates are needed.
